src/components/NewGalaxyModal.js

```diff
@@ -37,19 +43,23 @@ export const NewGalaxyModal = _.flow(
 )(({ onDismiss, onSuccess, apps, galaxyDataDisks, workspace, workspace: { workspace: { namespace, bucketName, name: workspaceName } } }) => {
   // Assumption: If there is an app defined, there must be a data disk corresponding to it.
   const app = currentApp(apps)
-  const dataDisk = currentDataDisk(app, galaxyDataDisks)
+  const attachedDataDisk = currentAttachedDataDisk(app, galaxyDataDisks)
 
-  const [dataDiskSize, setDataDiskSize] = useState(dataDisk?.size || defaultDataDiskSize)
+  const [dataDiskSize, setDataDiskSize] = useState(attachedDataDisk?.size || defaultDataDiskSize)
   const [kubernetesRuntimeConfig, setKubernetesRuntimeConfig] = useState(app?.kubernetesRuntimeConfig || defaultKubernetesRuntimeConfig)
   const [viewMode, setViewMode] = useState(undefined)
   const [loading, setLoading] = useState(false)
+  const [deleteDiskSelected, setDeleteDiskSelected] = useState(false)
```

Comment:
This name threw me off until I saw usage. No need to fix this, but you might consider whether something like `shouldDeleteDisk` or `shouldDeleteDiskSelected` would make the intent more obvious. Sorry for the extra noise, but this comment got deleted from my earlier review.

Reply:
no worries! will update